plutus-core/testlib/PlutusCore/Generators/Hedgehog/Denotation.hs

```diff
@@ -28,7 +28,7 @@ import Data.Dependent.Map qualified as DMap
 import Data.Functor.Compose
 import Type.Reflection
 
-type KnownType val a = (KnownTypeAst (UniOf val) a, MakeKnown val a)
+type KnownType val a = (KnownTypeAst TyName (UniOf val) a, MakeKnown val a)
```

Comment:
ditto

Reply:
were we just implicitly pinned to `TyName` somehow before, and this just makes it explicit? or what?

Reply:
> were we just implicitly pinned to `TyName` somehow before, and this just makes it explicit?

Yes, exactly. It used to be

```
class KnownTypeAst uni x where
    <...>
    toTypeAst :: proxy x -> Type TyName uni ()
    <...>
```

Now we're generic over `tyname`, but only for the very specific purpose of allowing `mkTyBulitin` to construct a `Type` with any `tyname` and so I get out of that generality as quickly as possible in order not to proliferate `tyname` through the entire code base. Like, `HasTypeLevel` propagates up to `PlutusTx`, we don't want to mention implementation details such as `TyName` there.